### DISCUSSION TITLE
Scriptability tool not working

### DISCUSSION THREAD

**Discussion by alexwillingham98** [thoughtspot/cs_tools#171] [General]
We run the scriptability tool weekly on a job and for some reason it quit working with this error:

│ C:\Users\VssAdministrator\AppData\Roaming\***\.***\Lib\site-packa │
│ ges\thoughtspot_tml\_tml.py:41 in recursive_complex_attrs_to_dataclasses    │
│                                                                             │
│    38 │   cls = type(instance)                                              │
│    39 │                                                                     │
│    40 │   for field in fields(cls):                                         │
│ >  41 │   │   value = getattr(instance, field.name)                         │
│    42 │   │                                                                 │
│    43 │   │   # ignore nulls, they get dropped to support optionality       │
│    44 │   │   if value is None:                                             │
│                                                                             │
│ C:\Users\VssAdministrator\AppData\Roaming\***\.***\Lib\site-packa │
│ ges\betterproto\__init__.py:840 in __getattribute__                         │
│                                                                             │
│    837 │   │   │   │   │   │   │   │   f"{group!r} is set to {group_current │
│    838 │   │   │   │   │   │   │   )                                        │
│    839 │   │   │   │   │   │   else:                                        │
│ >  840 │   │   │   │   │   │   │   raise AttributeError(                    │
│    841 │   │   │   │   │   │   │   │   f"{group!r} is set to {group_current │
│    842 │   │   │   │   │   │   │   │   name=name,                           │
│    843 │   │   │   │   │   │   │   │   obj=self,                            │
└─────────────────────────────────────────────────────────────────────────────┘
AttributeError: 'value_restrictions' is set to None, not 'list_config'         

┌──────────────────── 🤓  This is an unhandled error!  🤓 ────────────────────┐
│                                                                             │
│ If you encounter this message more than once, please help by letting us     │
│ know!                                                                       │
│                                                                             │
│           GitHub: https://github.com/thoughtspot/***/issues/new/choose │
│                                                                             │
└─────────── Run *** logs report to send us your last error. ────────────┘

**Comment by boonhapus** [comment]
Hey @alexwillingham98 - does your TML contain parameters?

If you're able, can you share your TML files with me directly?
https://thoughtspot.egnyte.com/ul/1bK0dz05L6

**Reply by alexwillingham98** [reply]
Yep, just sent

**Reply by boonhapus** [reply]
@alexwillingham98 I'm seeing no issue here. My guess is your `thoughtspot_tml` version is out of date.

```python
╰─❯ python
Python 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from thoughtspot_tml import utils
>>> import pathlib
>>> p = pathlib.Path(r"C:\Users\Downloads\b1de23c8-430b-4ed8-ba9f-0f6024e2d969.worksheet.tml")
>>> p.exists()
True
>>> utils.
utils.Answer(                        utils.TMLError(                      utils.fields(
utils.Connection(                    utils.TYPE_CHECKING                  utils.is_dataclass(
utils.EnvironmentGUIDMapper(         utils.Table(                         utils.json
utils.Liveboard(                     utils.View(                          utils.log
utils.MissingGUIDMappedValueWarning( utils.Worksheet(                     utils.logging
utils.Pinboard(                      utils.annotations                    utils.pathlib
utils.SQLView(                       utils.determine_tml_type(            utils.warnings
utils.TMLDisambiguationError(        utils.disambiguate(
>>> utils.determine_tml_type(path=p)
<class 'thoughtspot_tml.tml.Worksheet'>
>>> utils.determine_tml_type(path=p).load(path=p)
Worksheet(guid='b1de23c8-430b-4ed8-ba9f-0f6024e2d969', worksheet=WorksheetEDocProto(name='...'
```

Can you try one of these options?

- run `cs_tools self sync` to try and upgrade the version of `thoughtspot_tml`
- upgrade to v1.5.10 with `powershell -c "IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py | python - --install"`

**Reply by alexwillingham98** [reply]
So i actually have this command on an azure devops job that spins up a vm so it should be getting the latest version each time

**Reply by alexwillingham98** [reply]
I ran the job again this morning and this command succeeded this time. Not sure what changes haha but it seemed to resolve itself

**Reply by boonhapus** [reply]
If it pulls the latest, then that would include the restriction above. 😄
